python/pylibcudf/pylibcudf/join.pyx

```diff
@@ -38,8 +39,11 @@ __all__ = [
     "mixed_left_semi_join",
 ]
 
-cdef Column _column_from_gather_map(cpp_join.gather_map_type gather_map, Stream stream):
+cdef Column _column_from_gather_map(
+    cpp_join.gather_map_type gather_map, Stream stream, DeviceMemoryResource mr=None
+):
     # helper to convert a gather map to a Column
+    mr = _get_memory_resource(mr)
```

Comment:
```suggestion
    cpp_join.gather_map_type gather_map, Stream stream, DeviceMemoryResource mr
):
    # helper to convert a gather map to a Column
```

It appears the caller always passes a non-None `mr` as `mr = _get_memory_resource(mr)` is called before hand

Reply:
Good catch, yeah we should require the mr arg here.